perf/change_point_detector.go

```diff
@@ -0,0 +1,74 @@
+package perf
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/evergreen-ci/cedar"
+	"io"
+	"net/http"
+)
+
+type Detector interface {
+	DetectChanges([]float64) ([]ChangePoint, error)
+}
+
+type ChangePoint struct {
+	Index     int
+	Algorithm Algorithm
+}
+
+type Algorithm struct {
+	Name          string
+	Version       string
+	Configuration map[string]interface{}
+}
+
+type signalProcessingClient struct {
+	execute    func(method, url string, body io.Reader) (*http.Response, error)
```

Comment:
this feels like too much indirection, this probably doesn't need to be a type, you can just have different detector implementations and do your dependency injection at a higher level.

Reply:
additionally would it be possible, as this gets refactored, to change the way we pass data into this operation from a reader into something easier to introspect?

Reply:
Originally I didn't have this on there at all, but it seemed Like I needed this available in order to pass a mock wrapping `http.NewRequest`/setting up base url + header/`http.DefaultClient.Do`. Is there a better way to mock the client?

Reply:
http clients are even worse than this. I wouldn't really mock things at the level of the client.